src/raster/rasterband.rs

```diff
@@ -852,38 +880,230 @@ pub struct CmykEntry {
+        }
+    }
+}
+
+impl From<&ColorEntry> for GDALColorEntry {
+    fn from(e: &ColorEntry) -> Self {
+        match e {
+            ColorEntry::Gray(e) => GDALColorEntry {
+                c1: e.g,
+                c2: 0,
+                c3: 0,
+                c4: 0,
+            },
+            ColorEntry::Rgba(e) => GDALColorEntry {
+                c1: e.r,
+                c2: e.g,
+                c3: e.b,
+                c4: e.a,
+            },
+            ColorEntry::Cmyk(e) => GDALColorEntry {
+                c1: e.c,
+                c2: e.m,
+                c3: e.y,
+                c4: e.k,
+            },
+            ColorEntry::Hls(e) => GDALColorEntry {
+                c1: e.h,
+                c2: e.l,
+                c3: e.s,
+                c4: 0,
+            },
+        }
+    }
+}
+
+// For more compact debug output, skip enum wrapper.
+impl Debug for ColorEntry {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ColorEntry::Gray(e) => e.fmt(f),
+            ColorEntry::Rgba(e) => e.fmt(f),
+            ColorEntry::Cmyk(e) => e.fmt(f),
+            ColorEntry::Hls(e) => e.fmt(f),
+        }
+    }
+}
+
+/// Color table for raster bands that use [`ColorInterpretation::PaletteIndex`] color interpretation.
+///
+///
+/// # Example
+///
+/// ```rust, no_run
+/// use gdal::{Dataset, Driver};
+/// use gdal::raster::{ColorEntry, ColorTable, PaletteInterpretation};
+/// # fn main() -> gdal::errors::Result<()> {
 ///
-/// This object carries the lifetime of the raster band that
-/// contains it. This is necessary to prevent the raster band
-/// from being dropped before the color table.
+/// // Open source multinomial classification raster
+/// let ds = Dataset::open("fixtures/labels.tif")?;
+///
+/// // Create in-memory copy to mutate
+/// let mem_driver = Driver::get_by_name("MEM")?;
+/// let ds = ds.create_copy(&mem_driver, "<mem>", &[])?;
+/// let mut band = ds.rasterband(1)?;
+/// assert!(band.color_table().is_none());
+///
+/// // Create a new color table for 3 classes + no-data
+/// let mut ct = ColorTable::new(PaletteInterpretation::Rgba);
+/// ct.set_color_entry(0, &ColorEntry::rgba(255, 255, 0, 255));
+/// ct.set_color_entry(1, &ColorEntry::rgba(0, 255, 255, 255));
+/// ct.set_color_entry(2, &ColorEntry::rgba(255, 0, 255, 255));
+/// ct.set_color_entry(255, &ColorEntry::rgba(0, 0, 0, 0));
+/// band.set_color_table(&ct);
+///
+/// // Render a PNG
+/// let png_driver = Driver::get_by_name("PNG")?;
+/// ds.create_copy(&png_driver, "/tmp/labels.png", &[])?;
+///
+/// # Ok(())
+/// # }
+/// ```
 pub struct ColorTable<'a> {
     palette_interpretation: PaletteInterpretation,
     c_color_table: GDALColorTableH,
+    /// If `true`, Rust is responsible for deallocating color table pointed to by
+    /// `c_color_table`, which is the case when instantiated directly, as opposed to
+    /// when read via [`RasterBand::color_table`].
+    rust_owned: bool,
```

Comment:
To restate from comment above: If created via `new`, Rust deallocates. If retrieved from a `RasterBand`, GDAL deallocates.

Reply:
I _think_ this looks fine, but I've been wrong before :sweat_smile:.